evals/api.py

```diff
@@ -24,25 +25,26 @@
 logger = logging.getLogger(__name__)
 
 
-class ReturnType(ABC):
-    def __init__(self, raw_data: Any, prompt: Any, metadata: Dict[str, Any] = None):
+class CompletionResult(ABC):
+    def __init__(self, raw_data: Any):
         self.raw_data = raw_data
-        self.prompt: str = prompt
-        self.completions: List[str] = self.extract_completions()
-        self.metadata = metadata if metadata is not None else {}
 
     @abstractmethod
-    def extract_completions(self) -> List[str]:
+    def get_completions(self) -> list[str]:
         pass
 
+    @cached_property
+    def completions(self) -> list[str]:
+        return self.get_completions()
```

Comment:
Ideally I'd like to have only one way to get the completions, could we just keep one of these?

Reply:
how about we changed `get_completions` to `_get_completions` so it is clear to use `completions` publically?

Reply:
otherwise do something like this:
```
class CompletionResult(ABC):
    def __init__(self, raw_data: Any):
        self.raw_data = raw_data

    @abstractmethod
    @cached_property
    def get_completions(self) -> list[str]:
        pass
```

Reply:
Let's just keep `get_completions` as it was originally, I'm worried it might be confusing to have it like that (and not worth the convenience). Sorry for thrash!